exec/src/test/java/com/palantir/gradle/utils/exec/GradleExecTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.gradle.utils.exec;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.gradle.api.Action;
+import org.gradle.api.provider.Provider;
+import org.gradle.process.ExecSpec;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+class GradleExecTest {
+
+    @TempDir
+    Path tempDir;
+
+    private GradleExec gradleExec;
+
+    @BeforeEach
+    void beforeEach() {
+        gradleExec = ProjectBuilder.builder().build().getObjects().newInstance(GradleExec.class);
+    }
+
+    @Test
+    void lazyExec_shouldReturnProvider() {
```

Comment:
let's keep our standard of having the test name be all `snake_case` english sentence. Rather than doing `grouping_somethingElse` use `@Nested class Grouping`.